ui/src/app/pages/login/account-creation/account-creation.tsx

```diff
@@ -0,0 +1,668 @@
+    {label: `Community Scientist (i.e. I am accessing AoU for independent research, unrelated to my
+        professional affiliation)`, value: 'scientist'},
+    {label: `Other (free text)`, value: 'freeText'}
+  ],
+  industryRole: [
+    {label: 'Research Assistant (pre-doctoral)', value: 'preDoctoral'},
+    {label: 'Research associate (post-doctoral; early/mid career)', value: 'postDoctoral'},
+    {label: 'Senior Researcher (PI/Team Lead)', value: 'teamLead'},
+    {label: 'Other (free text)', value: 'freeText'}
+  ],
+  eductionRole: [
+    {label: 'Teacher/Professor', value: 'teacher'},
+    {label: 'Student', value: 'student'},
+    {label: 'Administrator', value: 'admin'},
+    {label: 'Other (free text)', value: 'freeText'}
+  ]
+};
+
+// The following will be part of Profile object once create account survey is in
+export interface Address {
+  streetAddress1: string;
+  streetAddress2: string;
+  city: string;
+  state: string;
+  zipcode: string;
+  country: string;
+}
+
+export interface AccountCreationProps {
+  invitationKey: string;
+  setProfile: Function;
+}
+
+export interface AccountCreationState {
+  creatingAccount: boolean;
+  errors: any;
+  invalidEmail: boolean;
+  rolesOptions: any;
+  profile: Profile;
+  showAllFieldsRequiredError: boolean;
+  showInstitution: boolean;
+  showAffiliationRole: boolean;
+  showAffiliationOther: boolean;
+  showNext: boolean;
+  usernameCheckInProgress: boolean;
+  usernameConflictError: boolean;
+  address: Address;
+  institutionName: string;
+  institutionRole: string;
+  affiliation: string;
+  affiliationRole: string;
+  affiliationOther: string;
+}
+
+const styles = {
+  sectionLabel: {
+    height: '22px',
+    color: colors.primary,
+    fontFamily: 'Montserrat',
+    fontSize: '16px',
+    fontWeight: 600,
+    lineHeight: '22px',
+    paddingBottom: '1.5rem'
+  },
+  section: {
+    width: '12rem'
+  }
+};
+
+const nameLength = 80;
+
+export const Section = (props) => {
+  return <FormSection
+      style={{display: 'flex', flexDirection: 'column', paddingTop: '1rem', ...props.style}}>
+    <label style={styles.sectionLabel}>
+      {props.header}
+    </label>
+    {props.children}
+  </FormSection>;
+};
+
+export class AccountCreation extends React.Component<AccountCreationProps, AccountCreationState> {
+  private usernameCheckTimeout: NodeJS.Timer;
+
+  constructor(props: AccountCreationProps) {
+    super(props);
+    this.state = {
+      errors: undefined,
+      profile: {
+        // Note: We abuse the "username" field here by omitting "@domain.org". After
+        // profile creation, this field is populated with the full email address.
+        username: '',
+        dataAccessLevel: DataAccessLevel.Unregistered,
+        givenName: '',
+        familyName: '',
+        contactEmail: '',
+        currentPosition: '',
+        organization: '',
+        areaOfResearch: '',
+      },
+      usernameCheckInProgress: false,
```

Comment:
I believe this state could be tracked in a more object-oriented way, for example by having a little component for each piece of data we're taking in and having it report errors (and be able to define what its own error state looks like). That way we could make them smarter later (e.g. allowing incremental feedback while you're typing, for characters remaining, forbidden characters, etc.). The way it is now, there's no real separation among all the input widgets' state, and the maintainer has to keep track fo whether each one is checked/flagged. 

This is just a quick look though; you could have reasons it needs to be centralize like this too.